go-controller/pkg/ovn/ovn.go

```diff
@@ -276,8 +279,55 @@ func (oc *Controller) Run(wg *sync.WaitGroup) error {
 	oc.WatchNodes()
 
 	oc.WatchPods()
-	oc.WatchServices()
-	oc.WatchEndpoints()
+
+	// Services are handled differently depending on the Kubernetes API versions
+	// We use a level triggered if k8s > 1.20, using endpoint slices instead endpoints
+	if util.DetectEndpointSlices(oc.client) {
+		// Services are handled differently depending on the Kubernetes API versions
+		klog.Infof("Using EndpointSlices instead of Endpoints in k8s versions > 1.19")
+		// Create our own informers to start compartamentalizing the code
+		// filter server side the things we don't care about
+		noProxyName, err := labels.NewRequirement("service.kubernetes.io/service-proxy-name", selection.DoesNotExist, nil)
```

Comment:
It would be nicer if this could go in the service controller code rather than having the generic init code need to know about the service controller's special cases...

Reply:
I'm not sure if I understand the comment, this is required to initialise the informer and filter server side.
This is how cmd/kube-proxy initialise the informer, Casey has done the same in the informer factory

Reply:
Yes, but it's weird to have `pkg/ovn/ovn.go` have intimate knowledge of how `pkg/ovn/controller/services/services_controller.go` wants its informer to be set up. It breaks the abstraction/modularization. It would be better if it just passed the client (or whatever) to the service controller, and then the service controller can create the informers however it wants them itself.

Reply:
ic your point  :+1: , 

but if we move to multiple controllers or we migrate from the current informerFactory code to `client-go`, where should we initialize all the needed informers?